src/core/internal/hash.d

```diff
@@ -64,14 +59,15 @@ if (!is(T == enum) && !is(T : typeof(null)) && is(T S: S[]) && !__traits(isStati
     static if (is(ElementType == interface) || is(ElementType == class) ||
                    (is(ElementType == struct) && !isNonReference!ElementType) ||
                    ((is(ElementType == struct) || is(ElementType == union))
-                       && is(typeof(val[0].toHash()) == size_t)))
+                       && is(typeof(val[0].toHash()) == size_t)) ||
+                   ((floatCoalesceZeroes || floatCoalesceNaNs) && __traits(isFloating, ElementType)))
     //class or interface array or struct array with toHash(); CTFE depend on toHash() method
     //also do this for arrays of structs whose hashes would be calculated in a memberwise fashion
     {
         size_t hash = seed;
         foreach (o; val)
         {
-            hash = hashOf(o, hash);
+            hash = hashOf(hashOf(o), hash); //double hashing because TypeInfo.getHash doesn't allow to pass seed value
```

Comment:
Please explain why this is necessary. Does this patch make `hashOf(o, hash)` not compile?

Reply:
Ah nevermind, I see
>Also make `typeid(T).getHash(&val)` to get the same result with `hashOf(val)`.